src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -647,9 +647,11 @@ private async ValueTask<bool> HandleSetName(ClientPlayer sender, string name)
 
                     if (name != expected)
                     {
-                        _logger.LogWarning($"Client sent {nameof(RpcCalls.SetName)} with incorrect name");
-                        await SetNameAsync(expected);
-                        return false;
+                        if (await sender.Client.ReportCheatAsync(RpcCalls.SetName, CheatCategory.GameFlow, "Client sent SetName with incorrect name"))
```

Comment:
Is this something that can never trigger legitimately? The loop above this is a bit difficult to read but I guess its to de-dupe duplicate names with a number suffix?

Reply:
I think that's correct, that should also be the behaviour of vanilla when it encounters two players with the same name. So vanilla should never trigger this